### PR TITLE
Add AMOs to CommDiagnostics.

### PR DESCRIPTION
**Pull request by gbtitus**
We've long wanted to include atomic operations in comm diagnostics.  Do
that here.  But param-disable printing atomic op counts by default,
because those counts are unstable due to our use of network atomics when
spin-waiting for parallel construct and on-stmt completion.

This is related to #10124, but doesn't resolve that because it doesn't enable
counting AMOs by default.

(Note that we would prefer to use a config const to control whether AMO
counts are printed, at execution time instead of compile time, but
initialization order constraints prevent using a config const here.
Basically, the problem is that the CommDiagnostics is initialized before
the String module is.)

### CODE REVIEW

File: modules/standard/CommDiagnostics.chpl [hunk 1]
```diff
@@ -222,10 +238,13 @@ module CommDiagnostics
...
       c <~> "(";
       for param i in 1..numFields(chpl_commDiagnostics) {
+        param name = getFieldName(this.type, i);
         const val = getField(this, i);
         if val != 0 {
-          if first then first = false; else c <~> ", ";
-          c <~> getFieldName(chpl_commDiagnostics, i) <~> " = " <~> val;
+          if commDiagsPrintUnstable || name != 'amo' {
+            if first then first = false; else c <~> ", ";
+            c <~> getFieldName(chpl_commDiagnostics, i) <~> " = " <~> val;
```
Comment (ronawho) [inline modules/standard/CommDiagnostics.chpl]: Replace now duplicate `getFieldName()` with `name`
